### ISSUE TITLE
NASNet

### ISSUE THREAD

**Issue by ahkarami** [Cadene/pretrained-models.pytorch#13]
Dear @Cadene,
Thank your for your nice repository. Recently the Google Brain Team released a fantastic CNN model, _[NASNet](https://arxiv.org/abs/1707.07012)_, in [TF-slim](https://github.com/tensorflow/models/tree/master/research/slim/nets/nasnet), which achieved the state-of-the-art Top-1 Accuracy on _ImageNet_ by **82.7 %**. Would you please kindly port this model into your nice repository (_i.e.,_ PyTorch Pre-Trained Models)?

**Comment by Cadene** [comment]
Hi @ahkarami,
I tried a bit this morning, but the tensorflow and slim API have changed a lot.... so I will try later with a friend who knows tensorflow better than me.
I will let you know.

**Comment by ahkarami** [comment]
Dear @Cadene,
Thank you for your time and response. I am also agree with your opinion about TensorFlow.

**Comment by Cadene** [comment]
@ahkarami I did not port the pretrained parameters yet to validate the model, but at least you can run a forward + backward pass on this version.
https://github.com/Cadene/pretrained-models.pytorch/blob/master/pretrainedmodels/nasnet.py

**Comment by Cadene** [comment]
Done ! https://github.com/Cadene/pretrained-models.pytorch/commit/247b0375abb6120ab4ef5642be59aee9d9e5542d

**Comment by qingzhouzhen** [comment]
Hi， have you verified the accuracy of the nasnet?, if yes, what is the train hyper-parameter? @Cadene

**Comment by Cadene** [comment]
@qingzhouzhen Here it is https://github.com/Cadene/pretrained-models.pytorch#accuracy-on-validation-set-single-model

For the train hyperparameters please look at the slim tensorflow repository

**Comment by qingzhouzhen** [comment]
OK, THX, I have found it in `https://github.com/tensorflow/models/blob/master/research/slim/nets/nasnet/nasnet.py`

**Comment by Cadene** [comment]
@qingzhouzhen What are the training hyperparameters then? :P

**Comment by qingzhouzhen** [comment]
`# Notes for training large NASNet model on ImageNet
# -------------------------------------
# batch size (per replica): 16
# learning rate: 0.015 * 100
# learning rate decay factor: 0.97
# num epochs per decay: 2.4
# sync sgd with 100 replicas
# auxiliary head loss weighting: 0.4
# label smoothing: 0.1
# clip global norm of all gradients by 10`
I found this in nasnet.py in Tensorflow, but it need 100GPUs, I am going to use batch parallel, and by the way, it said it has a `# auxiliary head loss weighting: 0.4`, and in TF'code, it write like this
`     #############################
      # Specify the loss function #
      #############################
      if 'AuxLogits' in end_points:
        slim.losses.softmax_cross_entropy(
            end_points['AuxLogits'], labels,
            label_smoothing=FLAGS.label_smoothing, weights=0.4,
            scope='aux_loss')
      slim.losses.softmax_cross_entropy(
          logits, labels, label_smoothing=FLAGS.label_smoothing, weights=1.0)
      return end_points`
how did you deal with this AuxLogits? I change your code to MXNET, batch can only arrive 8 in K80,

**Comment by ggsonic** [comment]
Thx for your great work! will the NASNet-A_Mobile_224 be supported? @Cadene

**Comment by Cadene** [comment]
@ggsonic Sorry I don't plan to do it at the moment. If you really need it, you should have a look at our porting code: https://github.com/Cadene/tensorflow-model-zoo.torch/tree/master/nasnet

Don't hesitate to contact me if you encounter any problem.

**Comment by ggsonic** [comment]
Thx! @Cadene  I will check it!

**Comment by Kulikovpavel** [comment]
Can you add NASNet medium?

According to this, it is best accuracy/complexity network

![image3](https://user-images.githubusercontent.com/1457767/50044358-4c314280-0093-11e9-8b98-43f7adb3771f.jpg)

https://ai.googleblog.com/2017/11/automl-for-large-scale-image.html
